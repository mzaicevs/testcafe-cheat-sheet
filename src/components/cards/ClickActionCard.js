import React from "react";
import { actions_click } from "../../snippets";
import CodeCard from "../../base_components/CodeCard";

const ClickActionCard = () => (
  <CodeCard
    cardTitle={actions_click.title}
    codeSnippet={{
      code: actions_click.code,
      language: actions_click.language,
      options: actions_click.options,
      example: actions_click.example
    }}
    description={actions_click.description}
  />
);

export default ClickActionCard;
