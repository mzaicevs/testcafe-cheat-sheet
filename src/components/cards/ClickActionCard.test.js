import React from "react";
import { shallow } from "enzyme";
import ClickActionCard from "./ClickActionCard";

jest.mock("react-syntax-highlighter/dist/esm/styles/hljs", () => () => {});

describe("ClickActionCard", () => {
  it("matches snapshot", () => {
    const wrapper = shallow(<ClickActionCard />);
    expect(wrapper).toMatchSnapshot();
  });
});
