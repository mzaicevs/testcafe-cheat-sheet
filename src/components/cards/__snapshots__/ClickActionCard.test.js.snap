// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClickActionCard matches snapshot 1`] = `
<CodeCard
  cardTitle={
    <CardLink
      href="https://devexpress.github.io/testcafe/documentation/test-api/actions/click.html"
      text="Click Action"
    />
  }
  codeSnippet={
    Object {
      "code": Array [
        "t.click( selector [, options] )",
        "t.doubleClick( selector [, options]",
        "t.rightClick( selector [, options] ))",
      ],
      "example": "import { Selector } from 'testcafe';

    const avatar = Selector('#avatar');
    const popupMenu = Selector('#popupMenu');
    
    fixture \`My fixture\`
        .page \`http://www.example.com/\`;
    
    test('Click a check box and check its state', async t => {
        await t
            .click(avatar)
            .expect(avatar.selected).ok();
            .doubleClick(avatar)
            .expect(avatar.visible).ok();
            .rightClick(avatar)
            .expect(popupMenu.exists).notOk();
    });",
      "language": "javascript",
      "options": "{
      modifiers: {
          ctrl: Boolean,
          alt: Boolean,
          shift: Boolean,
          meta: Boolean
      },
  
      offsetX: Number,
      offsetY: Number,
      destinationOffsetX: Number,
      destinationOffsetY: Number,
      speed: Number
  }",
    }
  }
  description="Click, Double-click, and Right-click a webpage element in a test."
/>
`;
